File: knowledgebase-sdk/src/main/java/ru/usedesk/knowledgebase_sdk/internal/di/InstanceBox.java
package ru.usedesk.knowledgebase_sdk.internal.di;

import android.content.Context;

import javax.inject.Inject;

import io.reactivex.annotations.NonNull;
import ru.usedesk.common_sdk.internal.appdi.InjectBox;
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase;
import ru.usedesk.knowledgebase_sdk.external.entity.KnowledgeBaseConfiguration;

@SuppressWarnings("injectable")
public class InstanceBox extends InjectBox {

    @Inject
    IUsedeskKnowledgeBase knowledgeBaseSdk;

    public InstanceBox(@NonNull Context appContext, @NonNull KnowledgeBaseConfiguration knowledgeBaseConfiguration) {
        init(new MainModule(appContext, knowledgeBaseConfiguration));
    }

    public IUsedeskKnowledgeBase getKnowledgeBaseSdk() {
        return knowledgeBaseSdk;
    }
}

File: knowledgebase-sdk/src/main/java/ru/usedesk/knowledgebase_sdk/internal/di/InstanceBox.kt
package ru.usedesk.knowledgebase_sdk.internal.di

import android.content.Context
import ru.usedesk.common_sdk.internal.appdi.InjectBox
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase
import ru.usedesk.knowledgebase_sdk.external.entity.UsedeskKnowledgeBaseConfiguration
import toothpick.ktp.delegate.inject

class InstanceBox(
        appContext: Context, knowledgeBaseConfiguration:
        UsedeskKnowledgeBaseConfiguration
) : InjectBox() {

    val knowledgeBaseSdk: IUsedeskKnowledgeBase by inject()

    init {
        init(MainModule(appContext, knowledgeBaseConfiguration))
    }
}